supports.js

```diff
@@ -210,28 +181,9 @@ window.matchMedia = window.matchMedia || (function (doc, undefined) {
 		},
 
 		mq: function (mq) {
-			if (window.matchMedia) {
-				return {
-					success: matchMedia(mq).media !== 'invalid'
-						? true
-						: matchMedia('not ' + mq).media !== 'invalid'
-				};
-			}
-			else {
-				style.textContent = '@media ' + mq + '{ foo {} }';
-
-				if (style.sheet.cssRules.length > 0) {
-					return {
-						success: true
-					};
-				} else {
-					style.textContent = '@media not ' + mq + '{ foo {} }';
-
-					return {
-						success: style.sheet.cssRules.length > 0 ? mq : false
-					};
-				};
-			}
+			return {
+				success: !matchMedia(mq).media.includes('not all')
```

Comment:
`includes` is used there instead of an equality check because according to the link you gave, when a browser does not recognize a media query, it is turned into `not all`, but the full media query list isn't.
So the intent was to avoid some false-positive results when testing comma-separated media queries, such as `layer, screen` which should be read as `not all, screen` and not just `not all`, but we don't really have such tests today (except the basic one for `print, screen` in *CSS 2 Media types*).
The other way to test the comma separator could indeed be a strick equality check and adding a test for something like `only, only` (which is invalid) because it should *not* be turned into `not all` only.
I wasn't sure which way to go.

I'll add a comment, sure.

Reply:
I've added the comment now.